src/brogue/Monsters.c

```diff
@@ -982,16 +1017,19 @@ boolean summonMinions(creature *summoner) {
     }
 
     if (summoner->info.abilityFlags & MA_ENTER_SUMMONS) {
-        if (atLeastOneMinion
-            && host) {
-
+        removeCreature(&monsters, summoner);
+        if (atLeastOneMinion && host) {
+            // TODO: is this right? This removes the summoner from the field,
+            // whererever it is that they are.
+            // TODO: what if there's more than one candidate?
+            // can this add the same summoner to multiple minions?
```

Comment:
1. Yes, I believe it's correct. The summer moves inside its summoned monster. Do you have any more details on why it might be wrong?
2. This code appears to be outside of any loop, so `host` will now be fixed from the loop above, to the last valid monster iterated over. Hence this code will only run once even when there are multiple candidates.

Reply:
Yeah, that sounds reasonable to me now. Something about the code previously made me suspicious but I think it was because I didn't understand something about how the chains were being used at the time.